query-planner-js/src/buildPlan.ts

```diff
@@ -195,23 +196,14 @@ const defaultCostFunction: CostFunction = {
   reduceDefer(nonDeferred: number, _: SelectionSet, deferredValues: number[]): number {
     return sequenceCost([nonDeferred, parallelCost(deferredValues)]);
   },
-
-  /**
-   * For roots, we just switch between with the sequence or parallel computation based on the type of root kind.
-   */
-  reduceRoots: (roots: number[], rootsAreParallel: boolean) => (
-    roots.length === 0
-      ? 0
-      : (rootsAreParallel ? parallelCost(roots) : sequenceCost(roots))
-  )
 };
 
 function parallelCost(values: number[]): number {
   return sum(values);
 }
 
 function sequenceCost(stages: number[]): number {
-  return stages.reduceRight((acc, stage, idx) => (acc + ((idx + 1) * pipeliningCost * stage)), 0);
+  return stages.reduceRight((acc, stage, idx) => (acc + (Math.max(1, idx * pipeliningCost) * stage)), 0);
```

Comment:
Note to reviewer: this change is due to the fact that the code now more or less rely on the fact that calling `processor.reduceSequence` on a single element is essentially a no-op, and that was true for the `reduceSequence` that generates the actual plan, but for not the costing of the plan here. Doesn't really change the overall logic otherwise.